tools/system_libs.py

```diff
@@ -286,325 +605,391 @@ def create_libc(libname):
-      files += map(lambda f: os.path.join(src_dir, f), filenames)
-    flags = ['-Oz']
-    flags += threading_flags(libname)
-    flags += legacy_gl_emulation_flags(libname)
-    flags += gl_version_flags(libname)
-    flags += gl_offscreen_flags(libname)
-    return build_libc(libname, files, flags)
-
-  # al
-  def create_al(libname): # libname is ignored, this is just one .o file
-    o = in_temp('al.o')
-    check_call([shared.PYTHON, shared.EMCC, shared.path_from_root('system', 'lib', 'al.c'), '-o', o, '-Os'] + get_cflags())
-    return o
-
-  def create_html5(libname):
-    src_dir = shared.path_from_root('system', 'lib', 'html5')
-    files = []
-    for dirpath, dirnames, filenames in os.walk(src_dir):
-      files += [os.path.join(src_dir, f) for f in filenames]
-    return build_libc(libname, files, ['-Oz'])
-
-  def create_compiler_rt(libname):
-    files = files_in_path(
-      path_components=['system', 'lib', 'compiler-rt', 'lib', 'builtins'],
-      filenames=['divdc3.c', 'divsc3.c', 'muldc3.c', 'mulsc3.c'])
-
-    o_s = []
-    commands = []
-    for src in files:
-      o = in_temp(os.path.basename(src) + '.o')
-      commands.append([shared.PYTHON, shared.EMCC, shared.path_from_root('system', 'lib', src), '-O2', '-o', o])
-      o_s.append(o)
-    run_commands(commands)
-    shared.Building.emar('cr', in_temp(libname), o_s)
-    return in_temp(libname)
-
-  # libc_extras
-  def create_libc_extras(libname): # libname is ignored, this is just one .o file
-    o = in_temp('libc_extras.o')
-    check_call([shared.PYTHON, shared.EMCC, shared.path_from_root('system', 'lib', 'libc', 'extras.c'), '-o', o] + get_cflags())
-    return o
-
-  # decides which malloc to use, and returns the source for malloc and the full library name
-  def malloc_decision():
-    if shared.Settings.MALLOC == 'dlmalloc':
-      base = 'dlmalloc'
-    elif shared.Settings.MALLOC == 'emmalloc':
-      base = 'emmalloc'
-    elif shared.Settings.MALLOC == 'none':
-      return (None, None)
-    else:
+    return libc_files
+
+  def get_depends(self):
+    depends = super(libc, self).get_depends()
+    if shared.Settings.WASM:
+      return depends + ['libc-wasm']
+    return depends
+
+
+class libc_wasm(MuslInternalLibrary):
+  name = 'libc-wasm'
+  symbols = read_symbols(shared.path_from_root('system', 'lib', 'wasm-libc.symbols'))
+
+  cflags = ['-O2', '-fno-builtin']
+  src_dir = ['system', 'lib', 'libc', 'musl', 'src', 'math']
+  src_files = ['cos.c', 'cosf.c', 'cosl.c', 'sin.c', 'sinf.c', 'sinl.c',
+               'tan.c', 'tanf.c', 'tanl.c', 'acos.c', 'acosf.c', 'acosl.c',
+               'asin.c', 'asinf.c', 'asinl.c', 'atan.c', 'atanf.c', 'atanl.c',
+               'atan2.c', 'atan2f.c', 'atan2l.c', 'exp.c', 'expf.c', 'expl.c',
+               'log.c', 'logf.c', 'logl.c', 'pow.c', 'powf.c', 'powl.c']
+
+  def can_use(self):
+    # if building to wasm, we need more math code, since we have less builtins
```

Comment:
```suggestion
    # if building to wasm, we need more math code, since we have fewer builtins
```

Reply:
This wasn't my comment either, but whatever.